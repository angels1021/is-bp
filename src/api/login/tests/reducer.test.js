/*
*
* Tests for login reducers
*
* */

import loginReducer from '../reducer';
import {
  requestSuccess,
  setAuth,
  sendingRequest,
  requestError,
  clearError
} from '../actions';

describe('Reducers for login api', () => {
  let state;
  beforeEach(() => {
    state = loginReducer(undefined, {});
  });

  it('should return the initial state', () => {
    expect(state).toMatchSnapshot();
  });

  describe('requestSuccess', () => {
    it('should handle the requestSuccess action correctly', () => {
      // arrange
      const expectedResult = state
        .mergeIn(['user'], {
          id: '1',
          uuid: '234',
          role: 'admin',
          firstName: 'Tom',
          lastName: 'Fletcher'
        });
      // act
      const result = loginReducer(state, requestSuccess({
        id: '1',
        uuid: '234',
        role: 'admin',
        firstName: 'Tom',
        lastName: 'Fletcher'
      }));
      // assert
      expect(expectedResult).toEqual(result);
    });
  });
  describe('setAuth', () => {
    it('should handle the setAuth action correctly', () => {
      // arrange
      const expectedResult = state.set('isLoggedIn', true);
      // act
      const result = loginReducer(state, setAuth(true));
      // assert
      expect(expectedResult).toEqual(result);
    });
  });

  describe('sendingRequest', () => {
    it('should handle the sendingRequest action correctly', () => {
      // arrange
      const expectedResult = state.set('pending', true);
      // act
      const result = loginReducer(state, sendingRequest(true));
      // assert
      expect(expectedResult).toEqual(result);
    });
  });

  describe('requestError', () => {
    it('should handle the requestError action correctly', () => {
      // arrange
      const expectedResult = state.set('error', 'exception');
      // act
      const result = loginReducer(state, requestError('exception'));
      // assert
      expect(expectedResult).toEqual(result);
    });
  });

  describe('clearError', () => {
    it('should handle the clearError action correctly', () => {
      // arrange
      const expectedResult = state.set('error', false);
      // act
      const result = loginReducer(state, clearError());
      // assert
      expect(expectedResult).toEqual(result);
    });
  });
});
