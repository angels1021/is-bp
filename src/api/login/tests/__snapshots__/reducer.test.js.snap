// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers for login api should return the initial state 1`] = `
Object {
  "error": false,
  "isLoggedIn": false,
  "pending": false,
  "user": Object {},
}
`;
